apps/contracts/scripts/test-cctp-adapter-transfer-remote.ts

```diff
@@ -0,0 +1,180 @@
+import {
+  DispatchedMessage,
+  HyperlaneCore,
+  MultiProvider,
+  chainMetadata,
+} from "@hyperlane-xyz/sdk";
+import { InterchainGasPaymaster__factory } from "@hyperlane-xyz/core";
+import { IERC20__factory } from "@hyperlane-xyz/core";
+import { sleep, addressToBytes32 } from "@hyperlane-xyz/utils";
+
+import { BigNumber, ethers } from "ethers";
+import yargs from "yargs";
+
+import { CctpAdapter__factory } from "../typechain-types";
+import cctpAdapters from "../artifacts/cctp-adapter-addresses.json";
+import cctpConfig from "../configs/cctp-adapter-deploy-config.json";
+import { getMultiProvider, assertBytes32 } from "../src/config";
+
+const mergeConfigs = (cctpConfig: any, cctpAdapters: any) => {
+  for (const chainId in cctpAdapters) {
+    cctpAdapters[chainId] = {
+      adapter: cctpAdapters[chainId],
+      usdc: cctpConfig[chainId].token,
+      igp: cctpConfig[chainId].interchainGasPaymaster,
+    };
+  }
+  return cctpAdapters;
+};
+
+const MIN_AMOUNT = BigNumber.from(1000); // 0.001 USDC
+
+async function transferRemote(
+  signer: ethers.Wallet,
+  core: HyperlaneCore,
+  multiProvider: MultiProvider,
+  addresses: any
+) {
+  // Need at least `transferAmount` on the first CctpAdapter contract defined in `artifacts/cctp-adapter-addresses.json`
+  const length = Object.keys(addresses).length;
+  const firstChainId = Object.keys(addresses)[0];
+  const usdc = IERC20__factory.connect(
+    addresses[firstChainId].usdc,
+    multiProvider.getProvider(firstChainId)
+  );
+  if (
+    (await usdc.balanceOf(signer.address)).lt(
+      BigNumber.from(MIN_AMOUNT).mul(length - 1)
+    )
+  ) {
+    throw new Error(`Insufficient funds on ${firstChainId}`);
+  }
+
+  // Distribute USDC to all chains first and use the distributed amount for testing transfer
+  const distributionAmount = MIN_AMOUNT.div(length - 1);
+  const transferAmount = distributionAmount.div(length - 1);
+  for (let i = 0; i < length; i++) {
+    const chainId = Object.keys(addresses)[i];
```

Comment:
Nit: I think you could do these loops with `for (const chainId of Object.keys(addresses)` which IMO reads better

Reply:
Done